dashboard/app/controllers/api/v1/teacher_feedbacks_controller.rb

```diff
@@ -74,6 +78,16 @@ def increment_visit_count
 
   private
 
+  def reset_progress_for_keep_working(teacher_feedback)
+    UserLevel.update_best_result(
+      teacher_feedback.student_id,
+      teacher_feedback.level_id,
+      teacher_feedback.script_id,
+      ActivityConstants::TEACHER_FEEDBACK_KEEP_WORKING,
+      false
```

Comment:
Not sure if this should be true or false.  I don't think we interpret `updated_at` anywhere so it's unclear what semantic meaning it has.  I think it could either be the last time that the _student_ did something to update progress or it could be the last that _any user_ did something to update progress or just the last time the row was touched for any reason.

Reply:
Oh..  does the comment [here](https://github.com/code-dot-org/code-dot-org/blob/4d5d5070912680e081141aacde35fe3d6f883738/dashboard/app/models/user_level.rb#L185) mean that we do assign a semantic meaning to this column?  Yikes.

Reply:
Yup, `updated_at` is supposed to represent the last time that the user worked on a level.

Reply:
Can we confirm with Amanda that resetting the progress in this way should  **not** update the last updated time shown to the student / teacher?

Reply:
I will add that the mocks show "Last updated by student <date>" which we wouldn't currently have a way of tracking if we set the updated_at date from a teacher action